src/unittest/test_irrptr.cpp

```diff
@@ -91,6 +91,11 @@ void TestIrrPtr::testRefCounting()
 			obj->getReferenceCount());
 }
 
+// GCC diagnostic works with clang, too
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wself-assign-overloaded"
+#pragma GCC diagnostic ignored "-Wself-move"
```

Comment:
I'm pretty sure you have to `#ifndef _MSC_VER` (or better `#if defined(__gcc__) || defined(__clang__)`) this